### DISCUSSION TITLE
Cannot get custom rules to work

### DISCUSSION THREAD

**Discussion by AlexxChanov** [detekt/detekt#5216] [Q&A] [answered]
Hi, I try to integrate the custom rule into my work project, but I can not do it because the project can not find my rule. So I was trying first to create a clean project to try to integrate the custom rule. What I did, I've found a working project in gitHub and found this one https://github.com/ChoChoHtet/detekt-custom-rules , I cloned it and tried to run detekt and the custom rules work fine there, so I created my new project and tried to do the same things as in project I've cloned. But I still can not to make the project find the custom rule. It just ignores it. Help me please to solve the problem. Here is my project: https://github.com/AlexxChanov/TestCustomRules

The project i've cloned show following:
```
equallines: 6
equallines: 1
equallines: 1
equallines: 1
/Users/alexchanov/AndroidStudioProjects/detekt-custom-rules/app/src/test/java/com/android/detekt_test/ExampleUnitTest.kt:3:1: Wildcard import [NoWildcardImports]
/Users/alexchanov/AndroidStudioProjects/detekt-custom-rules/app/src/main/java/com/android/detekt_test/MainActivity.kt:6:1: Importing 'kotlinx.android.synthetic.main.activity_main.txt_testing' which is a Kotlin Synthetics import. [NoSyntheticImport]
/Users/alexchanov/AndroidStudioProjects/detekt-custom-rules/app/src/main/java/com/android/detekt_test/MainActivity.kt:23:33: Change to single line [SingleExpressionFunction]
```

So we can see that rules are working here `SingleExpressionFunction` and `NoSyntheticImport` and it shows logs:
```
equallines: 6
equallines: 1
equallines: 1
equallines: 1
```

But in my project it doesn't show my custom rules:
```
> Task :customRules:compileJava NO-SOURCE
> Task :customRules:processResources UP-TO-DATE
> Task :customRules:classes UP-TO-DATE
> Task :customRules:inspectClassesForKotlinIC
> Task :customRules:jar

> Task :app:detekt
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

BUILD SUCCESSFUL in 9s
5 actionable tasks: 4 executed, 1 up-to-date 
```
I tried to copy all configurations, gradle version, detekt version, and so on, but it doesn't work in my project.
What did I do wrong?

**Comment by 3flex** [comment]
I'm moving this to the Discussions area, since this doesn't appear to be a bug at this stage.

**Comment by schalkms** [comment, marked as answer]
The error seems like a dependency problem.
https://stackoverflow.com/questions/7421612/slf4j-failed-to-load-class-org-slf4j-impl-staticloggerbinder

As a side note, usually, we suggest to go with the official [GitHub template](https://github.com/detekt/detekt-custom-rule-template) to have a basic scaffolding for developing 3rd party rules. Another option is to clone the provided [detekt/detekt-sample-extensions project](https://github.com/detekt/detekt/tree/main/detekt-sample-extensions).
